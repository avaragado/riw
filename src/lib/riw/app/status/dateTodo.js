// @flow

import path from 'path';

import compose from 'ramda/src/compose';

import type { Config } from '../../../config';

import dateMTimeFromFabs from './dateMTimeFromFabs';

type Result = (config: Config) => {
    dateTodo: number,
};

const result: Result = compose(
    dateTodo => ({ dateTodo }),
    dateMTimeFromFabs,
    (config: Config) => path.resolve(
        config.rootDir,
        config.todoFile,
    ),
);

export default result;
