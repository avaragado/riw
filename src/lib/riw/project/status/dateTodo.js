// @flow

import path from 'path';

import compose from 'ramda/src/compose';

import dateMTimeFromFabs from './dateMTimeFromFabs';

type Result = (config: RIWConfig) => {
    dateTodo: number,
};

const result: Result = compose(
    dateTodo => ({ dateTodo }),
    dateMTimeFromFabs,
    (config: RIWConfig) => path.resolve(
        config.rootDir,
        config.todoFile,
    ),
);

export default result;
