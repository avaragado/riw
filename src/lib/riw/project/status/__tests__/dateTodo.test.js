// @flow

import mock from 'mock-fs';

import dateTodo from '../dateTodo';
import cfgBase from '../../../__tests__/helpers/dummyConfig';

const cfgExtra = {
    ...cfgBase,
    rootDir: '.',
    todoFile: 'fixtures/dir/TODO.json',
};

type Fixture = {
    name: string,
    in: {
        [key: string]: any,
    },
};

const fixtures: Fixture[] = [
    {
        name: '01',
        in: {},
    },
    {
        name: '02',
        in: {
            'TODO.json': mock.file({
                content: '[]',
                mtime: new Date(2000),
            }),
        },
    },
];

describe('lib/riw/project/status/dateTodo', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            mock({
                fixtures: {
                    dir: fixture.in,
                },
            });

            const received = dateTodo(cfgExtra);

            mock.restore();

            expect(received).toMatchSnapshot();
        });
    });
});
