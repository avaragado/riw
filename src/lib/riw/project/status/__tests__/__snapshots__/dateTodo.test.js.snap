// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/riw/project/status/dateTodo 01 1`] = `
Object {
  "dateTodo": 0,
}
`;

exports[`lib/riw/project/status/dateTodo 02 1`] = `
Object {
  "dateTodo": 1970-01-01T00:00:02.000Z,
}
`;
